src/plugins/data/common/search/search_source/extract_references.ts

```diff
@@ -6,7 +6,7 @@
  * Side Public License, v 1.
  */
 
-import { SavedObjectReference } from '@kbn/core/types';
+import { SavedObjectReference } from '@kbn/core/server';
```

Comment:
Is it fine to import server types from common directory?

Reply:
Pretty surprised to see no complaint from eslint. IIRC, you shouldn't be able to import server or browser code into a common context unless it's a pure type import (`import type`) (cc @mistic )

Reply:
I'm going to check with @elastic/kibana-core before merging

Reply:
```suggestion
import type { SavedObjectReference } from '@kbn/core/server';
```

This should be fine, for safety we should explicitly mark as type only import